common-api/src/test/java/org/pmiops/workbench/db/model/CommonStorageEnumsTest.java

```diff
@@ -4,103 +4,86 @@
-        (Function<DataAccessLevel, Short>) CommonStorageEnums::dataAccessLevelToStorage
-      },
-      {
-        Domain.class.getSimpleName(),
-        Domain.values(),
-        (Function<Short, Domain>) CommonStorageEnums::domainFromStorage,
-        (Function<Domain, Short>) CommonStorageEnums::domainToStorage
-      }
-    };
-  }
-
-  @Parameter() public String description;
-
-  @Parameter(1)
-  public Enum<?>[] enumValues;
-
-  @Parameter(2)
-  public Function<Short, Enum<?>> fromStorage;
-
-  @Parameter(3)
-  public Function<Enum<?>, Short> toStorage;
-
-  @Test
-  public void testBijectiveStorageMapping() {
-    for (Enum<?> v : enumValues) {
-      Short storageValue = toStorage.apply(v);
-      assertWithMessage("unmapped enum value: " + v).that(storageValue).isNotNull();
-      assertThat(v).isEqualTo(fromStorage.apply(storageValue));
-    }
-  }
 
   // domain ID is stringly-typed so special-case this
 
   @Test
-  public void testDomainIdBijectiveStorageMapping() {
+  public void test_domainId() {
     for (Domain v : Domain.values()) {
       String storageValue = CommonStorageEnums.domainToDomainId(v);
       assertWithMessage("unmapped enum value: " + v).that(storageValue).isNotNull();
       assertThat(v).isEqualTo(CommonStorageEnums.domainIdToDomain(storageValue));
     }
   }
 
-  // copied from api/StorageEnumsTest because the above tests are not comprehensive
-  @Test
-  public void noMissingMapEntries() throws Exception {
-    for (Field f : CommonStorageEnums.class.getDeclaredFields()) {
-      if (f.getType() != BiMap.class) {
-        continue;
-      }
-
-      Class enumClass =
-          (Class) ((ParameterizedType) f.getAnnotatedType().getType()).getActualTypeArguments()[0];
-
-      Method enumToShort = null;
-      Method shortToEnum = null;
-      for (Method m : CommonStorageEnums.class.getDeclaredMethods()) {
-        if (m.getParameterTypes()[0].equals(enumClass)) {
-          enumToShort = m;
-        }
-
-        if (m.getReturnType().equals(enumClass)) {
-          shortToEnum = m;
-        }
-      }
-
-      // stringly typed map - test with testDomainIdBijectiveStorageMapping instead
-      if (enumToShort.getName().equals("domainIdToDomain")) {
-        continue;
-      }
+  // this is a copy of StorageEnumsTest.test_noMissingMapEntries()
 
+  @Test
+  public void test_noMissingMapEntries() throws Exception {
+    final Set<Class> enumClasses = getEnumerationClasses();
+    final Collection<Method> methods = Arrays.asList(CommonStorageEnums.class.getDeclaredMethods());
+
+    // e.g. public static Short domainToStorage(Domain domain)
+    final Map<Class, Method> toStorageMethods =
+        methods.stream()
+            .filter(m -> enumClasses.contains(firstMethodParameter(m)))
+            // domainToDomainId is stringly typed - test with test_domainId
+            .filter(m -> !m.getName().equals("domainToDomainId"))
```

Comment:
This test is extremely tightly tied to the implementation of the class. Ideally we capture enough of the functionality that migrating to something cleaner is easier. If we hold on too tightly, this could actually slow us down a bit.